Book-eCommerce-Store/Controllers/UsersController.cs

```diff
@@ -0,0 +1,87 @@
+using Book_eCommerce_Store.DTOs.Users;
+using Book_eCommerce_Store.Models;
+using Book_eCommerce_Store.Services.UsersService;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Book_eCommerce_Store.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class UsersController : ControllerBase
+    {
+        private readonly IUsersService _userService;
+
+        public UsersController(IUsersService userService)
+        {
+            _userService = userService;
+        }
+
+
+        [HttpGet]
+        public async Task<ActionResult<List<GetUserDTO>>> Get()
+        {
+            var response = await this._userService.Get();
+            if (response.Success == true)
+            {
+                response.Message = "Http Status OK";
+                return Ok(response);
+            }
+            return StatusCode(StatusCodes.Status500InternalServerError, response);
+
+        }
+
+        [HttpGet("{id}")]
+        public async Task<ActionResult<GetUserDTO>> GetById(int id)
+        {
+            var response = await this._userService.GetById(id);
+            if (response.Success == true)
+            {
+                response.Message = "Http Status OK";
+                return Ok(response);
+            }
+            return StatusCode(StatusCodes.Status500InternalServerError, response);
+        }
+
+        [HttpPost]
+        public async Task<ActionResult<GetUserDTO>> CreateUser(CreateUserDTO newUser)
+        {
+            var response = await this._userService.CreateUser(newUser);
+            if (response.Success == true)
+            {
+                response.Message = "Http Status OK";
+                return Ok(response);
+            }
+            return StatusCode(StatusCodes.Status500InternalServerError, response);
+        }
+
+        [HttpPut("{id}")]
```

Comment:
Since you can do a partial update on a User we should change from the HTTP PUT request method to the HTTP PATCH request method.